rmw_fastrtps_shared_cpp/src/utils/net_utils.cpp

```diff
@@ -0,0 +1,113 @@
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include "rcutils/logging_macros.h"
+#include "rmw_fastrtps_shared_cpp/utils/net_utils.hpp"
+
+namespace rmw_fastrtps_shared_cpp
+{
+namespace utils
+{
+std::string get_fqdn_for_host(const std::string& hostname) {
+  struct addrinfo hints, *addresses = nullptr;
+
+  memset(&hints, 0, sizeof(hints));
```

Comment:
memset is always a bit of a red flag for me, is there a better (more idiomatic C++ way) to initialize this? I've had to deal with `memset_s` and other issues in the past when considering safety standards, which is annoying, but for that reason I try to avoid it.

Reply:
One way is to use brace initialization. I'm not sure how portable it is though. Do any of these structs have `#ifdef ... int some_platform_specific_field;`?

```c++
struct addrinfo hints{0, 0, 0, 0, 0, 0, 0, 0};
```

Reply:
Speaking of which, does this even work on Windows?